### PR TITLE
file/swf: Use lzma-rs decompression instead of libhtp.

### PR DESCRIPTION
**Pull request by cccs-rtmorti**
Use the lzma-rs crate for decompressing swf/lzma files instead of the lzma decompressor in libhtp. This decouples suricata from libhtp except for actual http parsing, and means libhtp no longer has to export a lzma decompression interface.

Ticket: #5638

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/5638) ticket: 5638

Describe changes:
- Create a lzma.rs file which exports a lzma decompression function interface.
- In util-file-swf-decompression.c, use this interface instead of the lzma decompression interface presented by libhtp.
- Update error types / constants to reflect lzma-rs error conditions / results.

Update to  #8154

### CODE REVIEW

File: rust/src/lzma.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+use lzma_rs::decompress::{Options, Stream};
```
Comment (victorjulien) [inline rust/src/lzma.rs:1]: can you add a copyright header here?

**Review by victorjulien — CHANGES_REQUESTED** [review]
minor feedback

File: rust/src/lzma.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+impl From<Error> for LzmaStatus {
+    fn from(e: Error) -> LzmaStatus {
+        match e {
+            Error::IoError(_) => LzmaStatus::LzmaIoError,
+            Error::HeaderTooShort(_) => LzmaStatus::LzmaHeaderTooShortError,
+            Error::LzmaError(e) => {
+                if e.to_string().contains("exceeded memory limit") {
+                    LzmaStatus::LzmaMemoryError
```
Comment (catenacyber) [inline rust/src/lzma.rs:23]: Thanks for this

**Review by catenacyber — APPROVED** [review]
Thanks for this nice work.

CI : ✅
Code : looks good to me, comments were taken into account
Commits segmentation : ok
Commit messages : nice
Git ID set : looks fine for me
CLA : should be fine
Redmine ticket : ok
Rustfmt : fine for new file
Tests : good enough for me with current ones
Dependencies added: MIT License is good

File: rust/src/lzma.rs (hunk 1 above)
Comment (cccs-rtmorti) [reply rust/src/lzma.rs:1]: Yup! Sorry I forgot it.

**Comment by cccs-rtmorti** [comment]
Updated in #8176

**Pull request closed without merging**